docs/containers-registries.conf.5.md

```diff
@@ -30,14 +30,18 @@ as well as among different namespaces/repositories within a registry.
 Given an image name, a single `[[registry]]` TOML table is chosen based on its `prefix` field.
 
 `prefix`: A prefix of the user-specified image name, i.e. using one of the following formats:
-  - _host_[`:`_port_]
-  - _host_[`:`_port_]`/`_namespace_[`/`_namespace_…]
-  - _host_[`:`_port_]`/`_namespace_[`/`_namespace_…]`/`_repo_
-  - _host_[`:`_port_]`/`_namespace_[`/`_namespace_…]`/`_repo_(`:`_tag|`@`_digest_)
+  - [`*.`]_host_[`:`_port_]
+  - [`*.`]_host_[`:`_port_]`/`_namespace_[`/`_namespace_…]
+  - [`*.`]_host_[`:`_port_]`/`_namespace_[`/`_namespace_…]`/`_repo_
+  - [`*.`]_host_[`:`_port_]`/`_namespace_[`/`_namespace_…]`/`_repo_(`:`_tag|`@`_digest_)
 
     The user-specified image name must start with the specified `prefix` (and continue
     with the appropriate separator) for a particular `[[registry]]` TOML table to be
-    considered; (only) the TOML table with the longest match is used.
+    considered; (only) the TOML table with the longest match is used. It can
+    also include wildcarded subdomains in the format `*.example.com` along with
+    optional host, port, namespace additions as mentioned above. The wildcard
```

Comment:
The _exact_ requirements haven't been written down, so I really don't know.  But I think that if we further nail down the syntax of prefix, we should validate it at loading time.

Do you see something specific that would speak against supporting namespace matches?

Reply:
I wrote something specific above :)

For me, the most important part is consistency with `policy.json`, and that one does not support such matches (and can’t easily without externally-visible API changes to `ImageReference`), and is strongly focused on simplicity, primarily to avoid confusion (which is more important with the policy, admittedly).

IIRC the original motivation was https://issues.redhat.com/browse/RFE-1001 , which doesn’t require namespace matches from a quick skim. OTOH I haven’t now taken the time track all of it _fully_ down, some other RFEs might have been consolidated in to that one.

---

It’s _possible_ to imagine a user benefiting from `*.host/namespace[/*]`, but… why would one organize DNS _and_ registries that way? It seems unlikely and difficult to manage.

We can always add support for that later, if it turned out to be necessary (or I could well be wrong and it’s already necessary now, sure).

Reply:
Sounds good to me. @lsm5, feel free to reach out in case you have questions.

Reply:
ack, will make this change. Thanks!